firestore-cache/src/main.js

```diff
@@ -0,0 +1,173 @@
+// [[start common_connector_code]]
+var cc = DataStudioApp.createCommunityConnector();
+
+function getAuthType() {
+  return cc
+    .newAuthTypeResponse()
+    .setAuthType(cc.AuthType.NONE)
+    .build();
+}
+
+function getConfig(request) {
+  var config = cc.getConfig();
+
+  config
+    .newTextInput()
+    .setId('zipcode')
+    .setName('Enter Zip Code')
+    .setPlaceholder('eg. 95054');
+
+  return config.build();
+}
+
+function getFields() {
+  var fields = cc.getFields();
+  var types = cc.FieldType;
+  var aggregations = cc.AggregationType;
+
+  fields
+    .newDimension()
+    .setId('zipcode')
+    .setName('Zip code')
+    .setType(types.TEXT);
+
+  fields
+    .newDimension()
+    .setId('date')
+    .setName('Date')
+    .setType(types.YEAR_MONTH_DAY);
+
+  fields
+    .newMetric()
+    .setId('temperature')
+    .setName('Temperature (F)')
+    .setType(types.NUMBER)
+    .setIsReaggregatable(false);
+
+  return fields;
+}
+
+function getSchema(request) {
+  return {
+    schema: getFields().build()
+  };
+}
+// [[end common_connector_code]]
+
+// [[start caching_implementation]]
+function getData(request) {
+  var requestedFields = getFields().forIds(
+    request.fields.map(function(field) {
+      return field.name;
+    })
+  );
+
+  var cacheUpdateNeeded = true;
+  var url = buildFirebaseUrl(request.configParams.zipcode);
+  var cache = firebaseCache('get', url);
```

Comment:
If we're using firebase as a cache, I think this code would be clearer if we had functions like `getFromCache`, `deleteFromCache`, and `putInCache` or alternatively a cache object with `.get`, `.delete`, and `.put` methods.